scaleway/resource_domain_record_test.go

```diff
@@ -0,0 +1,592 @@
+package scaleway
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	domain "github.com/scaleway/scaleway-sdk-go/api/domain/v2beta1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+var (
+	// prevent using production domain for testing
+	reservedDomains = []*regexp.Regexp{
+		regexp.MustCompile(`.*iliad.*`),
+		regexp.MustCompile(`.*\.free\..*`),
+		regexp.MustCompile(`.*\.online\..*`),
+		regexp.MustCompile(`.*scaleway.*`),
+		regexp.MustCompile(`.*dedibox.*`),
+	}
+	testDomain = ""
+)
+
+func init() {
+	// if domain is set, use it
+	testDomainPtr := flag.String("test-domain", os.Getenv("TF_TEST_DOMAIN"), "Test domain")
+	if testDomainPtr != nil && *testDomainPtr != "" {
+		testDomain = *testDomainPtr
+		l.Infof("start domain record test with set domain: %s", testDomain)
+
+		return
+	}
+
+	// retrieve the first domain is found
+	scwClient, err := sharedClientForZone(scw.Zone(""))
+	if err != nil {
+		l.Errorf("error when retrieve scwClient: %s", err)
+		return
+	}
+
+	domainRegistrarAPI := domain.NewRegistrarAPI(scwClient)
+	listDomains, err := domainRegistrarAPI.ListDomains(&domain.RegistrarAPIListDomainsRequest{
+		Status:     domain.DomainStatusActive,
+		OrderBy:    domain.ListDomainsRequestOrderByDomainAsc,
+		IsExternal: scw.BoolPtr(true),
+	})
+	if err != nil {
+		l.Errorf("error listing domains: %s", err)
+		return
+	}
+
+	for _, domain := range listDomains.Domains {
```

Comment:
Please be careful with the naming. this variable `domain` collides with the an imported package